### PR TITLE
Share flags between chpl and chpldoc.

### PR DESCRIPTION
**Pull request by thomasvandoren**
Update chpl and chpldoc command line arguments to share a number of
flags, including:

* -h, --help, --help-env, --help-settings
* --version, --copyright, --license
* --[no-]devel, --print-chpl-home
* --gdb, --lldb

Previously, --help was the only common flags, and it was implemented in
two different ways.

In addition to flag sharing, update chpldoc to support setting author based
on command line flag and environment variable. The flag is --author and
the associated env var is CHPLDOC_AUTHOR. This is passed all the way
down to the sphinx project, which sets the copyright/author text based on
this value. Part of this functionality was added in #1713.

Add new commonFlags.h/cpp files, which manage the global flags (e.g.
fPrintHelp, fRunlldb, etc) and are #included by driver.cpp and
docsDriver.cpp. The arguments that are used by arg_desc and docs_arg_desc
are macros defined in commonFlags.h. This allows changes to these flags
to happen in one place and affect both chpl and chpldoc.

Remove the docs related flags (i.e. `fDocs*`) from driver.h. Update the
various files that use those flags to #include docsDriver.h. This includes
the lexer, so new flex output files are included.

Update chpldoc man page and test to include new options.

Add tests for the new flags on chpldoc. These reuse the helper scripts
in compflags/bradc/printstuff/ that are used to verify the same flags on
chpl.

### Verification:

* [x] code review
* [x] full test pass

### CODE REVIEW

**Comment by thomasvandoren** [comment]
This change doesn't add much that a chpldoc user (non-developer) would need. It does provide --version, --copyright, etc, but those are hardly mission critical and --help/-h is already supported on chpldoc.

This change does change driver.h and chapel.lex, which affects many other parts of the compiler. It also shuffles some of the global flags around. In theory, any issues would be realized at compile time, but that's not certain.

Given this change doesn't add much for non-developer chpldoc users, the fact that it changes some pretty critical files, and that developers will be working off master (i.e. the people who benefit most from it will not care if it is in the release or not), **I propose this be merged after the 1.11 release branch is cut.**

File: compiler/main/docsDriver.cpp [hunk 1]
```diff
@@ -17,10 +17,12 @@
  * limitations under the License.
  */
 
+#include "commonFlags.h"
+#include "driver.h"
 #include "docsDriver.h"
```
Comment (noakesmichael) [inline compiler/main/docsDriver.cpp]: I believe it's a good practice to ensure that the #include for the current implementation file appears first i.e.
#include "docsDriver.h" should appear first in this case.  This ensures that the header file does not accidentally
rely on any .h files that happen to be included first elsewhere.  This in turn makes it less likely that a change in
one header file will ripple unpredictably across the system.

File: compiler/passes/docs.cpp [hunk 2]
```diff
@@ -294,8 +295,9 @@ void generateSphinxOutput(std::string sphinxDir, std::string outputDir) {
   const char * venvBinDir = astr(venvDir, "/bin");
   const char * sphinxBuild = astr(venvBinDir, "/sphinx-build");
 
-  const char * envVars = astr("export PATH=", venvBinDir, ":$PATH && ",
-                              "export VIRTUAL_ENV=", venvDir);
+  const char * envVars = astr("export PATH=", venvBinDir, ":$PATH && \
+                              export VIRTUAL_ENV=", venvDir, " && \
+                              export CHPLDOC_AUTHOR='", fDocsAuthor, "'");
```
Comment (noakesmichael) [inline compiler/passes/docs.cpp]: I had forgotten this approach to splitting literal strings across multiple lines.  I am fine with this but wondered if you knew that the C compiler will automatically concatenate adjacent literal strings?

**Comment by noakesmichael** [comment]
I think this is ready to merge once the #include change has been made.

Thanks,
Mike

File: compiler/main/docsDriver.cpp (hunk 1 above)
Comment (thomasvandoren) [inline compiler/main/docsDriver.cpp]: I didn't know about this convention, but it makes sense. I updated this instance and the one in docs.cpp. Thanks!

File: compiler/passes/docs.cpp (hunk 2 above)
Comment (thomasvandoren) [inline compiler/passes/docs.cpp]: I was not aware of that. Thanks!

**Pull request merged**